### PR TITLE
Add state sampler.

### PR DESCRIPTION
**Pull request by sergeisakov**

### CODE REVIEW

**Comment by 95-martin-orion** [comment]
To read from file in the Bazel test, you'll need to include `circuit_q24` as a data dependency in the BUILD rule. See qsim_base in the `apps` BUILD file for an example of this.

File: lib/statespace_avx.h [hunk 1]
```diff
@@ -111,6 +112,57 @@ struct StateSpaceAVX final : public StateSpace<ParallelFor, float> {
         Base::num_threads_, Base::raw_size_ / 16, f, Op(), state);
   }
 
+  std::vector<uint64_t> Sample(
```
Comment (95-martin-orion) [inline lib/statespace_avx.h]: Does this make use of any AVX instructions? If not, I think we should provide a shared sampler for all statespace implementations.

Comment (95-martin-orion) [reply lib/statespace_avx.h]: My concern here isn't different floating-point types (although that's also a valid issue) - it's that statespace_basic has no equivalent sampling method.

Comment (sergeisakov) [reply lib/statespace_avx.h]: No, this doesn't make use of any AVX instructions. However, it does make use of specific state vector layout that is relevant only for the AVX version.

Comment (sergeisakov) [reply lib/statespace_avx.h]: I will add a sampler to statespace_basic in the next PR.

**Pull request merged**